src/fsharp/service/ServiceParsedInputOps.fsi

```diff
@@ -19,6 +19,7 @@ type public RecordContext =
     | CopyOnUpdate of range: range * path: CompletionPath
     | Constructor of typeName: string
     | New of path: CompletionPath
+    | Declaration of isInIdentifier: bool
```

Comment:
Should I remove the field and return `Declaration` only when the position is outside of the identifier, like in `UnionCaseFieldsDeclaration`?

Reply:
A more general question would be, should there be cases for different kinds of contexts even when those contexts end up with no completions,. Reusing `CompletionContext.Invalid` would yield the same result.

Reply:
Can I please get an answer here? This is the only thing that's holding up the PR as far as I am concerned.

Reply:
I don't have a strong intuition about this.  Could you get the PR green and then we can do a final end-to-end review?  It looks very thorough and a great contribution. thanks